dataverse-webapp/src/test/java/edu/harvard/iq/dataverse/dataset/difference/LicenseDifferenceFinderTest.java

```diff
@@ -0,0 +1,264 @@
+                .thenReturn(subjectFieldType.getControlledVocabularyValue("arts_and_humanities"));
+
+        when(datasetFieldService.findByNameOpt(eq("depositor"))).thenReturn(depositorFieldType);
+        when(datasetFieldService.findByNameOpt(eq("dateOfDeposit"))).thenReturn(dateOfDepositFieldType);
+
+
+        dataset = MocksFactory.makeDataset();
+    }
+
+    @Test
+    public void create__WITH_CHANGED_FILE_TERMS() throws IOException, JsonParseException {
+
+        // given
+
+        DataFile dataFile1 = MocksFactory.makeDataFile();
+        dataFile1.getFileMetadatas().clear();
+        DataFile dataFile2 = MocksFactory.makeDataFile();
+        dataFile2.getFileMetadatas().clear();
+
+        DatasetVersion v1 = parseDatasetVersionFromClasspath("/json/complete-dataset-version.json");
+        v1.setDataset(dataset);
+
+        FileMetadata v1FileMetadata1 = buildFileMetadata(10L, "firstFile.txt", 0, dataFile1);
+        FileTermsOfUse v1File1Terms = buildLicenseTermsOfUse(80L, "CCO0", "Public Domain", v1FileMetadata1);
+        FileMetadata v1FileMetadata2 = buildFileMetadata(11L, "secondFile.txt", 1, dataFile2);
+        FileTermsOfUse v1File2Terms = buildLicenseTermsOfUse(81L, "Apache 2.0", "Apache License", v1FileMetadata2);
+
+        v1.addFileMetadata(v1FileMetadata1);
+        v1.addFileMetadata(v1FileMetadata2);
+
+
+        DatasetVersion v2 = parseDatasetVersionFromClasspath("/json/complete-dataset-version.json");
+        v2.setDataset(dataset);
+
+        FileMetadata v2FileMetadata1 = buildFileMetadata(10L, "firstFile.txt", 0, dataFile1);
+        FileTermsOfUse v2File1Terms = buildLicenseTermsOfUse(81L, "Apache 2.0", "Apache License", v2FileMetadata1);
+        FileMetadata v2FileMetadata2 = buildFileMetadata(11L, "secondFile.txt", 1, dataFile2);
+        FileTermsOfUse v2File2Terms = buildRestrictedTermsOfUse(FileTermsOfUse.RestrictType.NOT_FOR_REDISTRIBUTION, v2FileMetadata2);
+
+        v2.addFileMetadata(v2FileMetadata1);
+        v2.addFileMetadata(v2FileMetadata2);
+
+
+        // when
+
+        List<DatasetFileTermDifferenceItem> termsDiffs = licenseDifferenceFinder.getLicenseDifference(v2.getFileMetadatas(), v1.getFileMetadatas());
+
+
+        // then
+
+        assertEquals(2, termsDiffs.size());
+
+        assertEquals(String.valueOf(dataFile1.getId()), termsDiffs.get(0).getFileSummary().getFileId());
+        assertSame(v1File1Terms, termsDiffs.get(0).getOldTerms());
+        assertSame(v2File1Terms, termsDiffs.get(0).getNewTerms());
+
+        assertEquals(String.valueOf(dataFile2.getId()), termsDiffs.get(1).getFileSummary().getFileId());
+        assertSame(v1File2Terms, termsDiffs.get(1).getOldTerms());
+        assertSame(v2File2Terms, termsDiffs.get(1).getNewTerms());
+    }
+
+    private DatasetVersion parseDatasetVersionFromClasspath(String classpath) throws IOException, JsonParseException {
+
+        try (ByteArrayInputStream is = new ByteArrayInputStream(IOUtils.resourceToByteArray(classpath))) {
```

Comment:
We can use UnitTestUtils for that now.

Reply:
What's that?

Reply:
It's just a wrapper for getting stuff from resources, we can also control which library we are going to use globally then.